internal/zinc-compile-core/src/main/scala/sbt/internal/inc/javac/LocalJava.scala

```diff
@@ -195,6 +186,30 @@ object LocalJava {
       }
     }
   }
+
+  private[sbt] def toUri(vf: VirtualFile): URI =
+    new URI(
+      "vf:/tmp/" + vf.id
+        .replaceAllLiterally("$", "%24")
+        .replaceAllLiterally("{", "%7B")
+        .replaceAllLiterally("}", "%7D")
+        .replaceAllLiterally(" ", "%20")
+    )
+
+  private[sbt] def fromUri(uri: URI): VirtualFileRef =
+    if (uri.getScheme != "vf") sys.error(s"invalid URI for VirtualFileRef: $uri")
+    else {
+      val part0 = uri.getSchemeSpecificPart
+      val part =
+        if (part0.startsWith("/tmp/")) part0.drop(5)
+        else part0
```

Comment:
That's `part0.stripPrefix("/tmp/")` btw.

Reply:
Good suggestion.